File: pkg/src/supportive/Console/Command/DebugUnassignedCommand.php
<?php

declare (strict_types=1);
namespace Qossmic\Deptrac\Supportive\Console\Command;

use Qossmic\Deptrac\Supportive\Console\Symfony\Style;
use Qossmic\Deptrac\Supportive\Console\Symfony\SymfonyOutput;
use DEPTRAC_INTERNAL\Symfony\Component\Console\Command\Command;
use DEPTRAC_INTERNAL\Symfony\Component\Console\Input\InputInterface;
use DEPTRAC_INTERNAL\Symfony\Component\Console\Output\OutputInterface;
use DEPTRAC_INTERNAL\Symfony\Component\Console\Style\SymfonyStyle;
class DebugUnassignedCommand extends Command
{
    public static $defaultName = 'debug:unassigned';
    public static $defaultDescription = 'Lists tokens that are not assigned to any layer';
    public const EXIT_WITH_UNASSIGNED_TOKENS = 2;
    public function __construct(private readonly \Qossmic\Deptrac\Supportive\Console\Command\DebugUnassignedRunner $runner)
    {
        parent::__construct();
    }
    protected function execute(InputInterface $input, OutputInterface $output) : int
    {
        $outputStyle = new Style(new SymfonyStyle($input, $output));
        $symfonyOutput = new SymfonyOutput($output, $outputStyle);
        try {
            $result = $this->runner->run($symfonyOutput);
            return $result ? self::EXIT_WITH_UNASSIGNED_TOKENS : self::SUCCESS;
        } catch (\Qossmic\Deptrac\Supportive\Console\Command\CommandRunException $exception) {
            $outputStyle->error($exception->getMessage());
            return self::FAILURE;
        }
    }
}

File: pkg/infrastructure/commands/debug_unassigned_command.go
package commands

import (
	services2 "github.com/KoNekoD/go-deptrac/pkg/application/services"
	"github.com/KoNekoD/go-deptrac/pkg/infrastructure/services/runners"
	"github.com/pkg/errors"
)

// DebugUnassignedCommand - debug:unassigned - Lists tokens that are not assigned to any layer
type DebugUnassignedCommand struct {
	runner *runners.DebugUnassignedRunner
}

func NewDebugUnassignedCommand(runner *runners.DebugUnassignedRunner) *DebugUnassignedCommand {
	return &DebugUnassignedCommand{runner: runner}
}

func (c *DebugUnassignedCommand) Run(output services2.OutputInterface) error {
	result, err := c.runner.Run(output)
	if err != nil {
		output.GetStyle().Error(services2.StringOrArrayOfStrings{String: "<fg=red>Unassigned token debugging failed.</> error: " + err.Error()})
		return errors.WithStack(err)
	}

	if result {
		output.WriteLineFormatted(services2.StringOrArrayOfStrings{String: "There are unassigned tokens."})

		return errors.New("There are unassigned tokens.")
	}

	return nil
}
